library/data/buffer/parser.lean

```diff
@@ -188,6 +199,22 @@ def fix_core (F : parser α → parser α) : ∀ (max_depth : ℕ), parser α
 | 0             := failure
 | (max_depth+1) := F (fix_core max_depth)
 
+/-- Matches a digit (0-9). -/
+def digit : parser nat := do
+  c ← any_char,
```

Comment:
```suggestion
  c ← sat (λ c, '0' ≤ c ∧ c ≤ '9'),
```

Then the error message will appear at the right position and backtracking will work as expected.

Reply:
Fixed, thanks!